relay/src/setup.rs

```diff
@@ -185,18 +164,36 @@ pub fn init_logging(config: &Config) {
     };
 
     let log = Box::new(log_builder.build());
-    let global_filter = log.filter();
-
-    sentry::integrations::log::init(
-        Some(log),
-        sentry::integrations::log::LoggerOptions {
-            global_filter: Some(global_filter),
-            attach_stacktraces: config.enable_backtraces(),
-            ..Default::default()
-        },
-    );
 
-    sentry::integrations::panic::register_panic_handler();
+    let log_integration = sentry::integrations::log::LogIntegration {
+        global_filter: Some(log.filter()),
+        attach_stacktraces: true,
+        dest_log: Some(log),
+        ..Default::default()
+    };
+
+    let guard = sentry::init(sentry::ClientOptions {
+        dsn: config
+            .sentry_dsn()
+            .map(|dsn| dsn.to_string().parse().unwrap()),
+        in_app_include: vec![
+            "relay_auth::",
+            "relay_common::",
+            "relay_config::",
+            "relay_filter::",
+            "relay_general::",
+            "relay_quotas::",
+            "relay_redis::",
+            "relay_server::",
+            "relay::",
+        ],
+        release: sentry::release_name!(),
+        integrations: vec![Arc::new(log_integration)],
+        ..Default::default()
+    });
+
+    // Keep the client initialized. The client is flushed manually in `main`.
```

Comment:
This is not important, but could we return the guard and drop it in main?

Reply:
Unfortunately not, because then we would drop the guard if we return an error. We could put the guard into a global and then take it in main, alternatively.

Reply:
My plan is to separate all logging and error-reporting from the CLI and setup code. In that case, we're infallible and can do what you suggest. That's a bigger change that needs to go in separately, though.